test/unit/hooks/sdtdlogs/index.test.js

```diff
@@ -71,4 +75,35 @@ describe('logging hook index', () => {
     expect(Player.update.args[0][1].country).to.equal('BE');
 
   });
+
+  describe('getLoggingObject', () => {
+
+    it('Returns an instance of LoggingObject', async () => {
+      const res = await sails.hooks.sdtdlogs.getLoggingObject(sails.testServer.id);
+      // TODO: This should be an instanceof assertion but JS classes are kicking me in the brain
```

Comment:
Yeah I have assertions, they aren't working yet :smile: Idk why but I couldn't get `instanceof` to recognize the returned object is a LoggingObject. Tests are failing because of this atm

Reply:
res.constructor == LoggingObject // false

thats why its not passing

Reply:
const res = new LoggingObject({ }); // await sails.hooks.sdtdlogs.getLoggingObject(sails.testServer.id);
makes it work fine

so my theory is what sails is doing to the hooks

Reply:
yea, so if you make getLoggingObject return `return new LoggingObject({ id: serverId });`, it fails again

Reply:
maybe just check to see if a function you care about is defined (ex check the shape of the object) instead of checking if its the exact same object.

Everything I can tell the two imports are not the same for no reason whatsoever. Probably something because its a hook in hooks directory and sails mass imports them and then wraps it with _.bindAll

Shape is probably better for your tests though, since you don't actually care if its the object

Reply:
![image](https://user-images.githubusercontent.com/110087/95909795-f17ed580-0d53-11eb-8f11-860a64c32af1.png)
infact adding the require after sails has lifted (the beforeAll) then works fine

Reply:
Thanks for this!!